app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx

```diff
@@ -0,0 +1,343 @@
+
+	useEffect(() => {
+		let timeoutId: NodeJS.Timeout;
+		if (isCopied) {
+			timeoutId = setTimeout(() => {
+				setIsCopied(false);
+			}, 2000);
+		}
+		return () => {
+			if (timeoutId) {
+				clearTimeout(timeoutId);
+			}
+		};
+	}, [isCopied]);
+
+	return (
+		<div className={cx(jenkinsSetupCopyButtonContainer)} >
+			<Button
+				iconBefore={<CopyIcon label="Copy" size="medium" />}
+				onClick={() => {
+					if (copyRef && copyRef.current) {
+						handleCopyToClipboard(copyRef);
+						setIsCopied(true);
+					}
+				}}
+				testId={testId}
+			>
+				Copy
+			</Button>
+
+			{isCopied ? <CopiedToClipboard /> : null}
+		</div>
+	);
+};
+
+type MyJenkinsAdminProps = {
+	webhookGuideRef: RefObject<HTMLDivElement>,
+	secretTokenRef: RefObject<HTMLDivElement>
+};
+
+const MyJenkinsAdmin = ({
+	handleCopyToClipboard,
+	webhookGuideRef,
+	secretTokenRef
+}: CopyProps & MyJenkinsAdminProps): JSX.Element => {
+	const tooltipContent =
+		'Send this secret token separately to the webhook URL and step-by-step guide. For example, if you used chat to send the webhook, send the secret via email.';
+
+	return (
+		<div className={cx(jenkinsSetupCopyContainer)}>
+			<p className={cx(jenkinsSetupContent)}>Copy the items below and give them to your Jenkins admin</p>
+
+			<ol className={cx(orderedList, jenkinsSetupOrderedList)}>
+				<li className={cx(orderedListItem, jenkinsSetupListItem)}>
+					Webhook URL and step-by-step guide
+					<CopyButton handleCopyToClipboard={handleCopyToClipboard} copyRef={webhookGuideRef} testId="copy-webhook-url-guide" />
+				</li>
+				<li className={cx(orderedListItem, jenkinsSetupListItem)}>
+					Secret token
+					<Tooltip content={tooltipContent} position="bottom-start">
+						<InfoIcon label="help" size="small" />
+					</Tooltip>
+					<CopyButton handleCopyToClipboard={handleCopyToClipboard} copyRef={secretTokenRef} testId="copy-secret-token-guide" />
+				</li>
+			</ol>
+
+			{/* TODO - convert info panel to a component */}
+			<div className={cx(jenkinsSetUpInfoPanel)}>
+				<PeopleGroup label="people-group" />
+				<div className={cx(jenkinsSetUpInfoPanelContentContainer)}>
+					<p>
+						Your Jenkins admin may need your input as they set up this server.&nbsp;
+						<InProductHelpAction
+							label="How to set up Jenkins servers to suit your team’s needs"
+							type={InProductHelpActionType.HelpLink}
+							appearance="link"
+							className="setUpInfoPanelHelpLink"
+						/>
+					</p>
+				</div>
+			</div>
+		</div>
+	);
+};
+
+type IAmTheJenkinsAdminProps = {
+	webhookUrlRef: RefObject<HTMLDivElement>,
+	secretRef: RefObject<HTMLDivElement>
+};
+
+const IAmTheJenkinsAdmin = ({
+	handleCopyToClipboard,
+	webhookUrlRef,
+	secretRef
+}: CopyProps & IAmTheJenkinsAdminProps): JSX.Element => {
+	const handleFollowLink = (e: React.MouseEvent): void => {
+		e.preventDefault();
+		router.open('https://app.contentful.com/spaces/zsv3d0ugroxu/entries/4ccCes4jpnMlSVtI4Eqre9?focusedField=body');
```

Comment:
/nit - not now, but might be worth creating consts up top for these contentful links, just so we know what they are and easier to find/replace

Reply:
lol I did actually do that in the other file where it's being used. Let me clean that up.